sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidOuterQueryRel.java

```diff
@@ -128,9 +129,14 @@ public DruidQuery toDruidQuery(final boolean finalizeAggregations)
       return null;
     }
 
+    final GroupByQuery groupByQuery = subQuery.toGroupByQuery();
+    if (groupByQuery == null) {
+      throw new CannotBuildQueryException("Subquery with limit clause is not supported");
```

Comment:
I don't think I understand the exception here. The error message talks about a subquery with a limit clause, but it looks like we're checking if the subQuery can not be converted to a group by query.

^ Sorry if this is a dumb question, I'm new here :)

Reply:
The exception message should be something like "Subquery could not be converted to GroupBy query", since it covers that broader case instead of just certain subqueries with limits.

Reply:
Indeed, it sounds better, thanks 👍